cmd/local-admin-client.go

```diff
@@ -80,3 +83,33 @@ func (lc localAdminClient) GetConfig() ([]byte, error) {
 
 	return json.Marshal(globalServerConfig)
 }
+
+// StartProfiling - starts profiling on the local server.
+func (lc localAdminClient) StartProfiling(profiler string) error {
+	if globalProfiler != nil {
+		globalProfiler.Stop()
+	}
+	globalProfiler = startProfiler(profiler, "")
+	return nil
+}
+
+// DownloadProfilingData - stops and returns profiling data of the local server.
+func (lc localAdminClient) DownloadProfilingData() ([]byte, error) {
+	profilerPath := globalProfiler.Path()
+
+	// Stop the profiler
+	globalProfiler.Stop()
+	globalProfiler = nil
```

Comment:
is it possible that running this code twice would simply crash the server since globalProfiler is nil ?

Reply:
> is it possible that running this code twice would simply crash the server since globalProfiler is nil ?

Absolutely, I forgot to fix this